aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -936,6 +941,62 @@ where
         })
     }
 
+    pub(crate) fn validate_data_reads(
+        &self,
+        data_map: &VersionedData<T::Key, T::Value>,
+        idx_to_validate: TxnIndex,
+    ) -> bool {
+        if self.non_delayed_field_speculative_failure {
+            return false;
+        }
+
+        // This includes AggregatorV1 reads and keeps BlockSTMv1 behavior intact.
+        self.validate_data_reads_impl(self.data_reads.iter(), data_map, idx_to_validate)
+    }
+
+    pub(crate) fn validate_aggregator_v1_reads(
+        &self,
+        data_map: &VersionedData<T::Key, T::Value>,
+        aggregator_write_keys: Option<impl Iterator<Item = T::Key>>,
+        idx_to_validate: TxnIndex,
+    ) -> Result<bool, PanicError> {
+        let mut aggregator_v1_iterable = Vec::with_capacity(self.aggregator_v1_reads.len());
```

Comment:
We should have only 1 aggregator in the system, so this should be fine to create a new vec I suppose?